### PR TITLE
Replace Marshal.SizeOf in System.Drawing.Common

### PR DESCRIPTION
**Pull request by nxtn**

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/EncoderParameter.cs [hunk 1]
```diff
@@ -154,7 +154,7 @@ public EncoderParameter(Encoder encoder, int numerator, int denominator)
 
             _parameterValueType = EncoderParameterValueType.ValueTypeRational;
             _numberOfValues = 1;
-            int size = Marshal.SizeOf(typeof(int));
+            int size = sizeof(int);
             _parameterValue = Marshal.AllocHGlobal(2 * size);
```
Comment (stephentoub) [inline src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/EncoderParameter.cs]: In cases like this, do we even need the local? I can't see what comes after it in the GitHub diff.

Comment (jkotas) [reply src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/EncoderParameter.cs]: We do. The EncoderParameter type has `[StructLayout(LayoutKind.Sequential)]` layout and it is used directly with managed interop.

File: src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/EncoderParameter.cs [hunk 2]
```diff
@@ -91,7 +91,7 @@ public EncoderParameter(Encoder encoder, byte value)
 
             _parameterValueType = EncoderParameterValueType.ValueTypeByte;
             _numberOfValues = 1;
-            _parameterValue = Marshal.AllocHGlobal(Marshal.SizeOf(typeof(byte)));
+            _parameterValue = Marshal.AllocHGlobal(sizeof(byte));
```
Comment (jkotas) [inline src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/EncoderParameter.cs]: Nit: You can also replace all `Marshal.WriteByte`s with direct pointer writes, e.g. `*(byte *)_parameterValue = value;`.

File: src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/EncoderParameter.cs [hunk 3]
```diff
@@ -91,7 +91,7 @@ public EncoderParameter(Encoder encoder, byte value)
...
 
             if (_parameterValue == IntPtr.Zero)
```
Comment (jkotas) [inline src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/EncoderParameter.cs]: `AllocHGlobal` never returns `IntPtr.Zero`. It throws `OutOfMemoryException` instead. You can delete all these `if (_parameterValue == IntPtr.Zero)` checks.

File: src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/EncoderParameter.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/EncoderParameter.cs]: I was referring to the `size` local.  Not sure if we're talking about the same thing.  If it is used, we should be able to make `int size = ...;` => `const int Size = ...;`

Comment (jkotas) [reply src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/EncoderParameter.cs]: Ah ok. You are right. This local should not be needed.

**Review by jkotas — APPROVED** [review]
Thank you!

**Pull request merged**